build_config/lib/src/build_config.dart

```diff
@@ -0,0 +1,344 @@
+// Copyright (c) 2017, the Dart project authors.  Please see the AUTHORS file
+// for details. All rights reserved. Use of this source code is governed by a
+// BSD-style license that can be found in the LICENSE file.
+
+import 'dart:async';
+import 'dart:io';
+
+import 'package:path/path.dart' as p;
+import 'package:yaml/yaml.dart';
+
+import 'pubspec.dart';
+
+/// The parsed values from a `build.yaml` file.
+class BuildConfig {
+  /// Supported values for the `platforms` attribute.
+  static const _allPlatforms = const [_vmPlatform, _webPlatform];
```

Comment:
Note that in bazel we now support 'flutter' as a platform as well I think

Reply:
@grouma for FYI.

Reply:
How should we keep this in sync with: https://github.com/dart-lang/rules_dart/blob/6bc0a5777089ec61bace0b3576b82c1edc0692e7/dart/build_rules/common/constants.bzl?

Reply:
We end up with tight version constraints between `dazel` and `rules_dart`. We'll have to keep the version of `build_config` in mind as well. We don't have a good story around preventing breakage because of things like this other than frequent testing.